frappe/public/js/frappe/form/footer/form_timeline.js

```diff
@@ -452,7 +452,7 @@ class FormTimeline extends BaseTimeline {
 		let content_wrapper = comment_wrapper.find('.content');
 
 		let delete_button = $();
-		if (frappe.model.can_delete("Comment")) {
+		if ((frappe.session.user == doc.owner && frappe.model.can_delete("Comment")) || frappe.session.user == 'Administrator') {
```

Comment:
As per @dottenbr's suggestion:

```suggestion
		if (frappe.model.can_delete("Comment") && (frappe.session.user == doc.owner || frappe.user.has_role("System Manager")) {
```

I think we should also add this validation to backend.